mesonbuild/linkers.py

```diff
@@ -982,9 +982,9 @@ class C2000DynamicLinker(DynamicLinker):
 
     id = 'cl2000'
 
-    def __init__(self, for_machine: mesonlib.MachineChoice,
+    def __init__(self, exelist: T.List[str], for_machine: mesonlib.MachineChoice,
                  *, version: str = 'unknown version'):
-        super().__init__(['cl2000.exe'], for_machine, '', [],
+        super().__init__(exelist or ['cl2000.exe'], for_machine, '', [],
```

Comment:
Is there any reason this shouldn't be `exelist or ['cl2000']`, since that should work for both windows and !windows?

Otherwise this looks good to me, if you could rebase this I'd be happy to merge it..

Reply:
Sure `exelist or ['cl2000']` should work for windows and non-windows OS but since this is a cross-compile setup, the compiler must be set within the crossfile and `exelist` will always be not empty, so the hard coded `cl2000.exe` isn't needed at all(?).